infra/scripts/qwen_audio_server.py

```
@@ -0,0 +1,287 @@
+            nested_url = audio_url.get("url")
+            if isinstance(nested_url, str):
+                return nested_url
+
+        raise HTTPException(
+            status_code=400,
+            detail="Audio content items must include a string `audio_url` field.",
+        )
+
+    def _load_audio(self, audio_url: str) -> Any:
+        try:
+            with urlopen(audio_url, timeout=30) as response:
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)